File: 2706-buy-two-chocolates/2706-buy-two-chocolates.cpp
class Solution {
public:
    int buyChoco(vector<int>& prices, int money) {
        int firstMin = INT_MAX, secondMin = INT_MAX;
        for(int &i: prices) {
            if(i<firstMin) {
                secondMin = firstMin;
                firstMin = i;
            } else if(i<secondMin) {
                secondMin = i;
            }
        }
        
        if(money-firstMin-secondMin<0)
            return money;
        
        return money-firstMin-secondMin;
    }
};

File: 2706-buy-two-chocolates/2706-buy-two-chocolates.go
func buyChoco(prices []int, money int) int {
    firstMin, secondMin := math.MaxInt, math.MaxInt
    
    for _,price := range prices {
        if firstMin>price {
            secondMin = firstMin
            firstMin = price
        } else if secondMin>price {
            secondMin = price
        }
    }
    
    if money-firstMin-secondMin<0 {
        return money
    }
    
    return money-firstMin-secondMin
}